File: 78-subsets/78. Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> a;
        for (int i = 0; i < pow(2, nums.size()); i++) {
            vector<int> x;
            for (int j = 0; j < nums.size(); j++) {
                if ((1 << j) & i) {
                    x.push_back(nums[j]);
                }
            }
            a.push_back(x);
        }
        return a;
    }
};

File: 78-subsets/78. Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        n = len(nums)
        for mask in range(1 << n):
            cur = []
            for i in range(n):
                if mask & (1 << i):
                    cur.append(nums[i])
            res.append(cur)
        return res